### ISSUE TITLE
deploy logs always display logs for existing pods

### ISSUE THREAD

**Issue by lnhrdt** [k14s/kapp#460]
**Issue**

I've noticed that for resources annotated with `kapp.k14s.io/deploy-logs`, kapp is printing logs for all of my new and existing pods. This is true both with the default value of `""` and with the specified valued `"for-new"`. According to the docs, both values should omit logs for existing pods and only show logs for new pods.

**Environment:**

- kapp: 0.46.0
- OS: alpine 14 (container)
- kubectl version: 1.21.6
- k8s cluster version: 1.21.6-gke.1503

---
Vote on this request

This is an invitation to the community to vote on issues, to help us prioritize our backlog. Use the "smiley face" up to the right of this comment to vote.

👍 "I would like to see this addressed as soon as possible"
👎 "There are other more important things to focus on right now"

We are also happy to receive and review Pull Requests if you want to help working on this issue.

**Comment by lnhrdt** [comment]
Here's an example resource (template) that is having this issue:

```yaml
---
apiVersion: batch/v1
kind: Job
metadata:
  name: tags-migrations
  labels:
    component: tags-migrations
  annotations:
    kapp.k14s.io/versioned: ""
    kapp.k14s.io/change-group: "migrations"
    kapp.k14s.io/change-rule: "upsert after upserting database"
spec:
  template:
    metadata:
      name: tags-migrations
      labels:
        component: tags-migrations
      annotations:
        kapp.k14s.io/deploy-logs: "for-new"
        kapp.k14s.io/deploy-logs-container-names: "tags-migrations"
    spec:
      restartPolicy: Never
      volumes:
        - name: migrations
          configMap:
            name: tags-migrations
      containers:
        - name: tags-migrations
          image: flyway/flyway
          args:
            - info
            - migrate
            - info
          env:
            - name: FLYWAY_URL
              valueFrom:
                configMapKeyRef:
                  name: database
                  key: DATABASE_CONNECTION_URL
            - name: FLYWAY_USER
              valueFrom:
                secretKeyRef:
                  name: migrations
                  key: DATABASE_USERNAME
            - name: FLYWAY_PASSWORD
              valueFrom:
                secretKeyRef:
                  name: migrations
                  key: DATABASE_PASSWORD
            - name: FLYWAY_SCHEMAS
              value: tags
          volumeMounts:
            - name: migrations
              mountPath: /flyway/sql
  backoffLimit: 0
```

**Comment by renuy** [comment]
@lnhrdt  adding this to our prioritized backlog. Thanks for bring this up

**Comment by renuy** [comment]
available as part v0.48.0
